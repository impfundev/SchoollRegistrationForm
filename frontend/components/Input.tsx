import * as React from "react";
import { View, TextInput, Text, StyleSheet } from "react-native";

import type { InputProps } from "@/types/InputProps";

const Input = React.forwardRef<any, InputProps>(
  (props, ref): React.ReactElement => {
    const { label, labelStyle, error, ...inputProps } = props;

    return (
      <View style={styles.container}>
        <Text style={styles.label}>{label && <Text>{label}</Text>}</Text>
        <TextInput
          style={styles.input}
          autoCapitalize="none"
          ref={ref}
          {...inputProps}
        />
        <Text>{error && error.message}</Text>
      </View>
    );
  }
);

const styles = StyleSheet.create({
  container: {
    display: "flex",
    gap: 8,
    justifyContent: "center",
    width: "100%",
  },
  label: {
    fontSize: 14,
    lineHeight: 14,
    fontWeight: "600",
  },
  input: {
    width: "100%",
    paddingLeft: 16,
    paddingRight: 16,
    paddingTop: 10,
    paddingBottom: 10,
    borderRadius: 9999,
    backgroundColor: "#fff",
    fontSize: 14,
    lineHeight: 14,
  },
});

export default Input;
