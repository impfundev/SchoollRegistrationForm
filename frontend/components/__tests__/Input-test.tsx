import * as React from "react";
import renderer from "react-test-renderer";
import Input from "../Input";

it(`renders correctly`, () => {
  const tree = renderer.create(<Input name="test" />).toJSON();
  expect(tree).toMatchSnapshot();
});
