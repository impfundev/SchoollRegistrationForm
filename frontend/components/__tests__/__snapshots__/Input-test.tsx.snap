// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "display": "flex",
      "gap": 8,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "fontSize": 14,
        "fontWeight": "600",
        "lineHeight": 14,
      }
    }
  />
  <TextInput
    autoCapitalize="none"
    name="test"
    style={
      {
        "backgroundColor": "#fff",
        "borderRadius": 9999,
        "fontSize": 14,
        "lineHeight": 14,
        "paddingBottom": 10,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 10,
        "width": "100%",
      }
    }
  />
  <Text />
</View>
`;
